tidb-binlog/driver/reader/reader.go

```diff
@@ -172,10 +183,12 @@ func (r *Reader) run() {
 				continue
 			}
 
-			r.msgs <- &Message{
+			msg = &Message{
 				Binlog: binlog,
 				Offset: kmsg.Offset,
 			}
```

Comment:
The problem here is that, after the caller of `reader` stops reading from `r.msgs`, this channel may become full and block this line and the whole `run` goroutine even after closed.

One way to avoid this is to add another `select` here with `<-r.stop` as a case and `continue` if `r.stop` is closed.

Reply:
the current solution also works, right?